Update readme to make supported code clearer

Firstyear opened the pull request:
Fixes https://github.com/rust-lang/miri/issues/1289

This makes it clearer in the readme how miri will warn of supported or unsupported code in a program under test/run. 

Thanks!

RalfJung commented on README.md:
````diff
@@ -100,6 +100,30 @@ fn does_not_work_on_miri() {
+    = note: inside call to `main` at /home/r/.rustup/toolchains/miri/lib/rustlib/src/rust/src/libstd/rt.rs:67:34
+```
+
+If you do not see an error like this, you are able to keep using `miri`!
````
TBH I think this spends way too much space on a problem that is mostly self-explaining when it comes up (as the error hopefully says everything you need to know). Sometimes saying too much about something is as confusing/distracting as saying too little about something. In particular "if you don't see an error, everything is all right" should go without saying.

My suggestion would be to simply amend "You can use this to ignore test cases that will fail under Miri because they do things Miri does not support:" with a parenthetical remark like:

> You can use this to ignore test cases that will fail under Miri because they do things Miri does not support (as indicated by "unsupported operation" errors when such an operation is executed):

Do you think that makes sense?

Firstyear replied:
Yeah, it's makes sense - I like my readme's and examples to be really explicit and very clear, but I will make the change.

Firstyear replied:
Sorry, that soundded rude, it wasn't my intent! I mean that I like my examples really detailed, and "exactly what a user will see" but I can make it more sparse :)

RalfJung commented:
Thanks for your PR! However, I don't think copying the full-blown error into such a prominent place in the README is a good use of that precious space (the longer we make that text the earlier we lose our readers). See the comment I added.

RalfJung commented:
@Firstyear  To make Github auto-close the issue when the PR lands, I edited the original description to say "fixes #1289".

Firstyear commented:
Okay, I've changed it to make the "error" example much smaller and focused on what lines to look for, and removed the extra sentence at the end. Is that better?

RalfJung commented:
Thanks! This looks good. I will do a bit more editing after this lands; I think that is easier than going back and forth.

@bors r+

bors commented:
:pushpin: Commit ef89525249614cd201aaa06d1652b693e9964c45 has been approved by `RalfJung`

<!-- @bors r=RalfJung ef89525249614cd201aaa06d1652b693e9964c45 -->
<!-- homu: {"type":"Approved","sha":"ef89525249614cd201aaa06d1652b693e9964c45","approver":"RalfJung"} -->

bors commented:
:hourglass: Testing commit ef89525249614cd201aaa06d1652b693e9964c45 with merge 5e7180bf469867375cfe6de957caa920f40ec488...
<!-- homu: {"type":"BuildStarted","head_sha":"ef89525249614cd201aaa06d1652b693e9964c45","merge_sha":"5e7180bf469867375cfe6de957caa920f40ec488"} -->

bors commented:
:sunny: Test successful - [checks-travis](https://travis-ci.com/github/rust-lang/miri/builds/158400498), [status-appveyor](https://ci.appveyor.com/project/rust-lang-libs/miri/builds/31953532)
Approved by: RalfJung
Pushing 5e7180bf469867375cfe6de957caa920f40ec488 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"RalfJung","base_ref":"master","builders":{"status-appveyor":"https://ci.appveyor.com/project/rust-lang-libs/miri/builds/31953532","checks-travis":"https://travis-ci.com/github/rust-lang/miri/builds/158400498"},"merge_sha":"5e7180bf469867375cfe6de957caa920f40ec488"} -->
